### PR TITLE
fix(epbs): always serve post-CL state for checkpoint sync, never FULL

### PR DESCRIPTION
**Pull request by lodekeeper**
## Summary

Fixes checkpoint state serving for post-Gloas finalized/justified lookups to always return the post-CL state (no payload applied). Ensures correct checkpoint sync behavior when epoch boundary slot 0 is missed.

## Problem

Two issues combined to produce 404s on checkpoint state lookups:

1. **Regen caching bug:** `processSlotsToNearestCheckpoint()` (called by `prepareNextSlot`) cached epoch boundary states using the head block's payload status (`FULL`). But `processSlots()` never processes payloads — the resulting state is always post-CL regardless of the source block's status.

2. **Missed slot edge case:** When epoch slot 0 is missed, `getCheckpointPayloadStatus()` correctly reports `EMPTY` (processSlot clears executionPayloadAvailability). The API forces `EMPTY` lookups, but only the `FULL` variant existed in cache → 404.

## Fix

**Regen** (`regen.ts`): Always cache post-Gloas epoch boundary states as `EMPTY` (`payloadPresent=false`). This is correct because `processSlots()` only does epoch transitions — never processes payloads.

**API** (`utils.ts`): For post-Gloas finalized/justified, force `EMPTY` lookup with **no fallback to FULL**. If EMPTY isn't found → 404 is the correct response. Serving a post-payload (FULL) state would break checkpoint sync for other clients. The regen fix guarantees the EMPTY variant is always cached.

## Testing

- `pnpm lint` ✅
- `pnpm check-types` ✅ (0 errors in beacon-node)
- **7 unit tests** covering:
  - `getStateValidatorIndex` — existing tests (restored)
  - `getStateResponseWithRegen` — 404 when only FULL variant exists (never serves post-payload state)
  - Post-Gloas missed-slot-0: verifies 404, not silent FULL fallback
  - Normal case (block at epoch boundary): EMPTY variant found on first try
  - Justified checkpoint: same EMPTY-only behavior
  - Regen caching: verifies EMPTY for Gloas epoch boundary states

## Context

- [ethereum/beacon-APIs#572](https://github.com/ethereum/beacon-APIs/issues/572) — potuz, sproul, twoeths agree finalized/justified should always serve post-CL state
- Discord #epbs discussion (2026-03-13): potuz flagged the missed-slot-0 edge case, barnabasbusa removed Lodestar from devnet checkpoint sync rotation

## Notes
AI-assisted investigation and patch.

### CODE REVIEW

File: packages/beacon-node/test/unit/api/impl/beacon/state/utils.test.ts [hunk 1]
```diff
@@ -1,104 +1,147 @@
...
-      // "validator pubkey not in state"
-      expect(
-        getStateValidatorIndex(
-          "0xa99af0913a2834ef4959637e8d7c4e17f0b63adc587d36ab43510452db3102d0771a4554ea4118a33913827d5ee80b76",
-          state,
-          pubkey2index
-        ).valid
-      ).toBe(false);
-    });
+describe("getStateResponseWithRegen — post-Gloas checkpoint serving", () => {
```
[the remark this answers is not in the document]
Comment (lodekeeper) [reply packages/beacon-node/test/unit/api/impl/beacon/state/utils.test.ts]: Restored in 6c4f185c78 — all original `getStateValidatorIndex` tests are back alongside the new checkpoint serving tests. Good catch @gemini-code-assist.

**Comment by lodekeeper** [comment]
Squash-merged by Nico in c3cd2bb5 with lint/type fixes + payloadPresent propagation through processSlotsByCheckpoint. All core changes intact — closing.
